java/BUILD.bazel

```diff
@@ -0,0 +1,187 @@
+        "@com_google_guava_guava//jar",
+    ],
+)
+
+java_library(
+    name = "org_ray_ray_test",
+    srcs = glob(["test/src/main/java/**/*.java"]),
+    deps = [
+        "@com_google_guava_guava//jar",
+        "@commons_collections_commons_collections//jar",
+        ":org_ray_ray_api",
+        ":org_ray_ray_runtime",
+        "@org_apache_commons_commons_lang3//jar",
+        "@org_slf4j_slf4j_api//jar",
+        "@plasma//:org_apache_arrow_arrow_plasma",
+        "@org_testng_testng//jar",
+    ],
+)
+
+java_library(
+    name = "org_ray_ray_runtime_test",
+    srcs = glob(["runtime/src/test/java/**/*.java"]),
+    deps = [
+        ":org_ray_ray_api",
+        ":org_ray_ray_runtime",
+        "@org_testng_testng//jar",
+        "@org_apache_commons_commons_lang3//jar",
+        "@com_beust_jcommander//jar",
+    ],
+)
+
+java_test(
+    name = "AllTests",
+    size = "small",
+    main_class = "org.testng.TestNG",
+    data = ["testng.xml"],
+    runtime_deps = [
+        ":org_ray_ray_test",
+        ":org_ray_ray_runtime_test",
+    ],
+)
+
+genrule(
+    name = "generate_java_gcs_fbs",
+    srcs = [":copy_java_gcs_fbs"],
+    outs = ["generate_java_gcs_fbs.out"],
+    cmd = """
+        python3 $$(pwd)/java/modify_generated_java_flatbuffers_files.py $$(pwd)
+        echo $$(pwd) > $@
+    """,
+    local = 1,
+)
+
+flatbuffer_java_library(
+    name = "java_gcs_fbs",
+    srcs = ["//:gcs_fbs_file"],
+    outs = [
+        "ActorCheckpointData.java",
+        "ActorCheckpointIdData.java",
+        "ActorState.java",
+        "ActorTableData.java",
+        "Arg.java",
+        "ClassTableData.java",
+        "ClientTableData.java",
+        "ConfigTableData.java",
+        "CustomSerializerData.java",
+        "DriverTableData.java",
+        "ErrorTableData.java",
+        "ErrorType.java",
+        "FunctionTableData.java",
+        "GcsTableEntry.java",
+        "HeartbeatBatchTableData.java",
+        "HeartbeatTableData.java",
+        "Language.java",
+        "ObjectTableData.java",
+        "ProfileEvent.java",
+        "ProfileTableData.java",
+        "RayResource.java",
+        "ResourcePair.java",
+        "SchedulingState.java",
+        "TablePrefix.java",
+        "TablePubsub.java",
+        "TaskInfo.java",
+        "TaskLeaseData.java",
+        "TaskReconstructionData.java",
+        "TaskTableData.java",
+        "TaskTableTestAndUpdate.java",
+    ],
+    out_prefix = "java/runtime/src/main/java/org/ray/runtime/generated",
+)
+
+genrule(
+    name = "copy_java_gcs_fbs",
+    srcs = [
+        "//java:java_gcs_fbs",
+    ],
+    outs = [
+        "java_gcs_fbs.out",
+    ],
+    cmd = """
+        WORK_DIR=$$(pwd)
+        GENERATED_DIR=$$WORK_DIR/java/runtime/src/main/java/org/ray/runtime/generated
+        mkdir -p $$GENERATED_DIR
+        for f in $(locations //java:java_gcs_fbs); do
+            cp $$f $$GENERATED_DIR/;
+        done
+        echo $$(pwd) >> $@
+    """,
+    local = 1,
```

Comment:
You already specify `out_prefix = "java/runtime/src/main/java/org/ray/runtime/generated"` above. Why need to copy again?

Reply:
> You already specify `out_prefix = "java/runtime/src/main/java/org/ray/runtime/generated"` above. Why need to copy again?

Because it didn't generate the files to the directory